cypress/integration/project_checkout.spec.js

```diff
@@ -0,0 +1,102 @@
+import { DIALOG_SHOWOPENDIALOG, OPEN_PATH } from "../../native/ipc.js";
+
+const withWorkspaceStub = callback => {
+  cy.exec("pwd").then(result => {
+    const pwd = result.stdout;
+    const checkoutPath = `${pwd}/cypress/workspace/checkout`;
+
+    // Clean up any left-overs from previous failed tests.
+    cy.exec(`rm -rf ${checkoutPath}`);
+    cy.exec(`mkdir ${checkoutPath}`);
+
+    // Stub Electron native calls
+    cy.window().then(appWindow => {
+      appWindow.mockOpenPathCalled = false;
+
+      appWindow.electron = {
+        ipcRenderer: {
+          invoke: msg => {
+            if (msg === DIALOG_SHOWOPENDIALOG) {
+              return checkoutPath;
+            } else if (msg === OPEN_PATH) {
+              appWindow.mockOpenPathCalled = true;
+            }
+          },
+        },
+      };
+    });
+
+    callback(checkoutPath);
+
+    // Clean up the cypress workspace.
+    cy.exec(`rm -rf ${checkoutPath}`);
+  });
+};
+
+beforeEach(() => {
+  cy.nukeAllState();
+  cy.createIdentity();
+  cy.createProjectWithFixture();
+  cy.visit("./public/index.html");
+});
+
+context("project checkout", () => {
+  context("happy path", () => {
+    it("checks out the project into a working directory", () => {
+      cy.pick("project-list-entry-platinum").click();
+      cy.pick("checkout-modal-toggle").click();
+
+      withWorkspaceStub(checkoutPath => {
+        cy.pick("choose-path-button").click();
+        // Make sure UI has time to update path value from stub,
+        // prevents this spec from failing on CI.
+        cy.wait(500);
```

Comment:
Is there no way to check the stub's condition or something?

Reply:
Not that I know of or wouldn't involve a lot of boilerplate, we have the same issue in project creation: https://github.com/radicle-dev/radicle-upstream/blob/master/cypress/integration/project_creation.spec.js#L270.